packages/studio-ui/src/utils/loadUserAssets.tsx

```diff
@@ -0,0 +1,62 @@
+import useStudioStore from "../store/useStudioStore";
+import getFunctionComponent from "./getFunctionComponent";
+import dynamicImportFromBrowser from "./dynamicImportFromBrowser";
+import { FileMetadataKind } from "@yext/studio-plugin";
+
+/**
+ * Load all of the user's components into the store.
+ */
+export function loadComponents(): Promise<void>[] {
+  const { UUIDToFileMetadata, setImportedComponent, UUIDToImportedComponent } =
+    useStudioStore.getState().fileMetadatas;
+
+  const componentImportPromises = Object.values(UUIDToFileMetadata).map(
+    async (fileMetadata) => {
+      if (UUIDToImportedComponent.hasOwnProperty(fileMetadata.metadataUUID)) {
+        return;
+      }
+      const importedModule = await dynamicImportFromBrowser(
+        fileMetadata.filepath
+      );
+      const functionComponent = getFunctionComponent(importedModule);
+      if (functionComponent) {
+        setImportedComponent(fileMetadata.metadataUUID, functionComponent);
+      }
+    }
+  );
+  return componentImportPromises;
+}
+
+/**
+ * Load all user styling as disabled style tags in Studio's head.
+ */
+export function loadStyling() {
+  const studioStore = useStudioStore.getState();
+  const UUIDToFileMetadata = studioStore.fileMetadatas.UUIDToFileMetadata;
+  const pages = studioStore.pages.pages;
+
+  Object.values(UUIDToFileMetadata).forEach((fileMetadata) => {
+    if (fileMetadata.kind === FileMetadataKind.Component) {
+      importAndInjectIntoStudio(fileMetadata.cssImports);
+    }
+  });
+
+  Object.values(pages).forEach((page) => {
+    importAndInjectIntoStudio(page.cssImports);
+  });
+}
+
+function importAndInjectIntoStudio(cssImports: string[]) {
+  cssImports.forEach(async (filepath) => {
+    await dynamicImportFromBrowser(filepath).then((styling) => {
+      if (document.querySelector(`[studio-style-filepath='${filepath}']`)) {
```

Comment:
Not sure I follow. Is the dynamic import needed in order to check if the style tag is a duplicate?

Reply:
oops I misread your comment!  You're right, I was concerned about some asynchronous cases where we might be dynamically importing the same component at the same time, but since we await the dynamic import I think that those concerns are void. I'll move it up.

Reply:
As we chatted offline about, I changed the stream to a for loop to prevent race conditions